compiler/Makefile

```diff
@@ -56,10 +56,12 @@ SUBDIRS = \
 #
 include $(COMPILER_ROOT)/make/Makefile.compiler.head
 
+ALL_HEADERS= */*.h dyno/include/chpl/*/*.h
```

Comment:
This didn't seem to fix the cases noted in #20379 for me—did they for you?  Unless I got my `git grep` wrong, I thought they were defined in `dyno/lib/parsing/*.h`?  (this is partially why I filed the issue rather than just trying to fix it myself... I wasn't confident I understood the dyno code structure well enough to get all the right paths).

And then as a housekeeping/code review comment: Given this approach, it might make sense to have this PR remove the `.h` portion of this addition to ALL_SRCS since it's redundant:

```
compiler/main/Makefile.include:ALL_SRCS += main/*.h main/*.cpp
```

(I realize that this is a pre-existing redundancy, so arguably not this PR's problem, so just an opportunity to housekeep if you're so motivated... I hadn't realized/remembered until seeing this diff that some headers were explicitly added here after noting that at least some were defined within `ALL_SRCS`).

Reply:
@bradcray - Ah, I hadn't noticed that you were trying to jump to types that were defined in private header files. (The dyno header file strategy is that dyno/include/chpl is the directory that would be installed somewhere i.e. forms the public API for the library; but within the lib subdirectories we can have private header files that are useful for the implementation).

I've updated this PR to also include those headers.

Reply:
That seems to work better.  I'm still not able to jump to a very satisfying place for the definition of `MaybeNamedActualList`, but that seems to be more due to issues related to indirecting from a bison-generated header to the bison file itself than anything missing in our tags command.  Thanks!